app/lib/dns.server.ts

```diff
@@ -0,0 +1,192 @@
+import {
+  Route53Client,
+  CreateHostedZoneCommand,
+  ChangeResourceRecordSetsCommand,
+  GetChangeCommand,
+} from "@aws-sdk/client-route-53";
+import { CreateHostedZoneResponse } from "@aws-sdk/client-route-53";
```

Comment:
This is a `type` import vs. a module.  You should separate it below your module imports with a blank line, and use `import type` vs. `import`:

```ts
import {
  Route53Client,
  CreateHostedZoneCommand,
  ChangeResourceRecordSetsCommand,
  GetChangeCommand,
} from "@aws-sdk/client-route-53";

import type { CreateHostedZoneResponse } from "@aws-sdk/client-route-53";
```

Also, do you not need the other response types as well?  That is, for all the other command types you are executing?  I think the issue is that `.send(command)` doesn't return the specific type, and you have to set it when declaring the response.